src/main/java/io/kokuwa/maven/helm/DependencyBuildMojo.java

```diff
@@ -28,6 +41,180 @@ public class DependencyBuildMojo extends AbstractHelmMojo {
+				dependency.setVersion(overwriteDependencyVersion);
+				dependency.setOverwrite(true);
+			}
+		}
+	}
+
+	/**
+	 * Overwrite the dependencies with their new values to the Chart.yaml 
+	 * @param dependencies {@link ArrayList}<{@link Dependency}>
+	 * @param chartFile Chart.yaml file to read
+	 * @param chartDirectory Directory to write the new Chart.yaml to
+	 * @throws MojoExecutionException
+	 */
+	private void overwriteDependencies(ArrayList<Dependency> dependencies, File chartFile, Path chartDirectory) 
+		throws MojoExecutionException {
+
+		//delete the outdated Chart.lock
+		File chartLock = new File(chartDirectory.toString() + "/Chart.lock");
+		
+		if (chartLock.exists()) {
+			chartLock.delete();
+			getLog().info("Chart.lock deleted at " + chartDirectory);
+		} else {
+			getLog().info("No Chart.lock file found at " + chartDirectory);
+		}
+
+		List<String> overwrittenChart = new ArrayList<>();
+		List<String> originalChart;
+		try {
+			originalChart = Files.readAllLines(Paths.get(chartFile.toURI()), StandardCharsets.UTF_8);
+		} catch (IOException e) {
+			throw new MojoExecutionException("Unable to read Chart.yaml at " + chartDirectory, e);
+		}
+
+		//Used for determining when a dependency has been read
+		boolean name, repository, version;
+		name = version = repository = false;
+
+		int currentDependencyIndex = 0;
+		Dependency currentDependency = dependencies.get(currentDependencyIndex);
+
+		for (String line : originalChart) {
+
+			//all dependencies have been read so just need to write remaining lines
+			if (currentDependencyIndex == dependencies.size()) {
+				overwrittenChart.add(line);
+				continue;
+			}
+
+			//Repository is not a required field so could be null
+			if (currentDependency.getRepository() == null) {
+				repository = true;
+			}
+
+			//check for lines that need to be overwritten
+			if (line.contains(" version:")) { 
+				version = true;
+				
+				if (currentDependency.isOverwrite()) {
+					getLog().info("Overwriting dependency '" + currentDependency.getName() + "' with new version:");
+					getLog().info("\tOld Value:");
+					getLog().info(line);
+
+					//preserve any characters/spacing that may appear before the attribute
+					int endIndex = line.indexOf(" version:") + "version:".length();
+					line = line.substring(0, endIndex + 1);
+					line += " " + overwriteDependencyVersion;
+
+					getLog().info("\tNew Value:");
+					getLog().info(line);
+				}
+			} else if (line.contains(" repository:")) {
+				repository = true;
+
+				if (currentDependency.isOverwrite()) {
+					getLog().info("Overwriting dependency '" + currentDependency.getName() + "' with new repository:");
+					getLog().info("\tOld Value:");
+					getLog().info(line);
+
+					//preserve any characters/spacing that may appear before the attribute
+					int endIndex = line.indexOf(" repository:") + "repository:".length();
+					line = line.substring(0, endIndex + 1);
+					line += " " + overwriteDependencyRepository;
```

Comment:
Q/A: What about trailing comments? It might be more straightforward to use `line.replace(currentDependency.getRepository(), overwriteRepository)`

Reply:
Didn't think of that scenario, that should work!

Reply:
Gonna have to create a new var to hold the original line value, but should still be doable

Reply:
Actaully I think we can just do `line.replace(currentDependency.getRepository(), overwriteRepository)` and that will avoid having to do the substring thing at all